keras/utils/layer_utils.py

```diff
@@ -200,6 +236,22 @@ def print_summary(model,
     positions.append(line_length)
     to_display.append('Trainable')
 
+  if layer_range is not None:
+    if len(layer_range) != 2:
```

Comment:
I think we probably want to push those validation logic into get_layer_index_bound_by_layer_name as well. They are duplicated here and in vis_util.

Reply:
Sure, having this freedom i also tried to simplify and remove some unnecessary validation logic by rewriting some lines in `get_layer_index_bound_by_layer_name` function.